src/engine/attributes/string.js

```diff
@@ -1,8 +1,11 @@
-import { convertToString, startsWith, endsWith } from "../../../src/utils/lang/index.js";
-
 export default function stringApply(val, userVal, op) {
+
+    if (userVal === undefined || userVal === null || userVal.constructor.name != "String") {
```

Comment:
We should check empty string as well.

Reply:
we allow empty string, same like other sdks.

Reply:
Hmm right. Is there any use case of allowing empty string?